### ISSUE TITLE
Issues in updating flutter from 3.19.6 to 3.22.0

### ISSUE THREAD

**Issue by emanuelzallikodelabs** [flutter/flutter#148664]
### Steps to reproduce

When upgrading Flutter from 3.19.6 to 3.22.0, the app just freezes in debug mode and release mode

### Expected results

I get this from Xcode: io.flutter.1.ui (10): EXC_BAD_ACCESS (code=1, address=0x12893ffd0)

### Actual results

I get this from Xcode: io.flutter.1.ui (10): EXC_BAD_ACCESS (code=1, address=0x12893ffd0)

### Code sample

<details open><summary>Code sample</summary>

```dart
[Paste your code here]
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.22.0, on macOS 14.5 23F79 darwin-arm64, locale en-XK)
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 15.4)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2023.3)
[✓] IntelliJ IDEA Ultimate Edition (version 2024.1.1)
[✓] Connected device (5 available)
    ! Error: Browsing on the local area network for Test’s iPad. Ensure the device is unlocked and attached with a cable or associated with the same local area network as this Mac.
      The device must be opted into Developer Mode to connect wirelessly. (code -27)
[✓] Network resources

**Comment by darshankawar** [comment]
@emanuelzallikodelabs 
You'll need to provide us complete runnable reproducible code sample (without third party implementation) that shows the error you are reporting. Without it, we will not be able to make the issue actionable.
Also provide us entire log in properly formatted text that shows meaningful entries that we can analyze further.

**Comment by darshankawar** [comment]
@emanuelzallikodelabs 
Please don't reply to comment via email, as it doesn't allow to format it. Also when you are posting the log, please format it so that it is helpful to be readable. Currently we can't gauge anything based on your reply. Please provide us proper error log in formatted text and also reproducible code that throws the reported behavior.

**Comment by Rr01010010** [comment]
It is very very very WEIRD. I has that error on flutter 3.22, when on earlier version 3.19.2 works stable.
I get this error on page with horizontal scrolling PageView two children as SingleChildScrollView

to be more exactly:
```
class SimplifiedExampleOfPage extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return PageView(
      controller: appLifeTimedRouteController.contentController,
      children: [
        SingleChildScrollView(
          controller: appLifeTimedRouteController.finalFirstFeedController,
          child: Column(children: []),
        ),
        SingleChildScrollView(
          controller: appLifeTimedRouteController.finalSecondFeedController,
          child: Column(children: []),
        )
      ],
    );
  }
}
```

Apparently, the ScrollController calls the dispose method, or some other destructor of some sort, when the scrollview is not in the screen's scope. And since I have two scrollviews used on the same page, it seems that some mechanisms for re-initializing the final scrollcontroller are not called, (this is my hypothesis). I tried to solve it by making the ScrollController static final - this caused interesting behavior, because if the ScrollView did not crash the application once, then in the case of a hot-restart, the application would no longer crash, no matter how many restarts occurred... It was as if the static controller was saved in memory. when changing final to mutable var there was no difference, it crashed in 80% of cases, as before with final

short error text:
```
* thread #9, name = 'io.flutter.1.ui', stop reason = EXC_BAD_ACCESS (code=1, address=0x11bae3fd0)
    frame #0: 0x0000000108592d54 Flutter`flutter::DisplayListBuilder::Restore() + 284
Flutter`flutter::DisplayListBuilder::Restore:
->  0x108592d54 <+284>: ldrsb  w8, [x24]
    0x108592d58 <+288>: sub    w8, w8, #0x17
    0x108592d5c <+292>: cmp    w8, #0x1
    0x108592d60 <+296>: b.hi   0x1085931e4               ; <+1452>
```

**I resolve that with refuse SingleChildScrollView, and rewrite the layout using ListView. This crutch solution solved my problem and allowed me to stabilize the customer’s application.**

**Comment by GuibertLo** [comment]
Same behaviour here, temporarily fixed it by:
1. Downgrading Flutter to version **3.19.6**
```
flutter downgrade 3.19.6
``` 
2. Forcing intl dependency to version 0.18.1 in project pubspec (following lien to force this version for all dependencies, [might cause issues](https://dart.dev/tools/pub/dependencies#dependency-overrides))
```
dependency_overrides:
  intl: 0.18.1
```

**Comment by Rr01010010** [comment]
> Same behaviour here, temporarily fixed it by:
> 
> 1. Downgrading Flutter to version **3.19.6**
> 
> ```
> flutter downgrade 3.19.6
> ```
downgrade flutter it's not decision. Refusal new version of flutter, just because scroll widget is sick, is a mistake.

**Comment by darshankawar** [comment]
Everyone in this thread facing the reported issue which we still don't know the exact behavior, please provide us steps to replicate and reproducible code sample that claims that the issue occurs after updating. 

I tried the given code sample above but it has custom implementation so we can't properly verify:

<img width="748" alt="Screenshot 2024-05-27 at 11 08 13 AM" src="https://github.com/flutter/flutter/assets/67046386/f2caaad6-61a7-4d6e-9e21-5f21817ec508">

**Comment by emanuelzallikodelabs** [comment]
Must be replaced on pages where .map() is used in ListView or ListView.builder

**Comment by darshankawar** [comment]
It is not clear what exactly above comment means @emanuelzallikodelabs Please provide concise runnable reproducible code sample so as to make this issue actionable rather than going back and forth.

**Comment by GuibertLo** [comment]
> > @GuibertLo
> > Same behaviour here, temporarily fixed it by:
> > 
> > 1. Downgrading Flutter to version **3.19.6**
> 
> downgrade flutter it's not decision. Refusal new version of flutter, just because scroll widget is sick, is a mistake.

Agreed, but if you have to publish an app quickly, it's better than nothing.

**Comment by Rr01010010** [comment]
@darshankawar 
This code, without creating a default MaterialApp, might be suitable for identifying the problem? Or is it necessary to write an ultra-typical material?

As I wrote above, the problem is not in some unique code. Page view and scroll view controllers use the most common controllers. The problem is that they behave inconsistently.

```
class SimplifiedExampleOfPage extends StatefulWidget {
  final PageController contentController = PageController();
  final ScrollController finalFirstFeedController = ScrollController();
  final ScrollController finalSecondFeedController = ScrollController();

  @override
  State<SimplifiedExampleOfPage> createState() =>
      _SimplifiedExampleOfPageState();
}

class _SimplifiedExampleOfPageState extends State<SimplifiedExampleOfPage> {
  final subwidgets = [
    Colors.yellow,
    Colors.blue,
    Colors.red,
    Colors.green,
    Colors.purple,
    Colors.cyan,
  ]
      .map(
        (color) => Container(
          decoration: BoxDecoration(
            borderRadius: BorderRadius.circular(15),
            color: color,
          ),
          padding: const EdgeInsets.all(20),
          height: 200,
          width: double.maxFinite,
        ),
      )
      .toList();

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Colors.white,
      body: PageView(
        controller: widget.contentController,
        children: [
          SingleChildScrollView(
            controller: widget.finalFirstFeedController,
            child: Column(children: subwidgets),
          ),
          SingleChildScrollView(
            controller: widget.finalSecondFeedController,
            child: Column(children: subwidgets),
          )
        ],
      ),
    );
  }
}
```
